### PR TITLE
Add Azure Monitor OpenTelemetry Exporter 

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
* Generated autogen code from swagger
* Added basic functionality to send data to azure monitor

### CODE REVIEW

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:6]: nit, put `OpenTelemetry` namespace below `System`.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:73]: Default to `200` seems scary, as future addition to the `StatusCanonicalCode` would default to it.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:153]: Minor: if we know the fields that we care about, we might want to list them here instead of iterating through the entire collection and do pattern matching - for sake of perf.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj:5]: Do we want this to be `1.0.0` or `0.0.1` 😄

**Review by reyang — APPROVED** [review]
I think this is good as an initial draft.

**Comment by reyang** [comment]
We probably want to update the [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/master/.github/CODEOWNERS) file at some point, @pakrym what's the general guidance here?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs
Comment (TimothyMothra) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs:14]: Please fill in these summary's before merging. We've had issues with devs forgetting.

**Comment by pakrym** [comment]
>We probably want to update the CODEOWNERS file at some point, @pakrym what's the general guidance here?

We have a guide about the forma here: https://github.com/Azure/azure-sdk/blob/master/docs/policies/opensource.md#codeowners

But in general, go for it.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj
Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj:5]: https://semver.org/#how-do-i-know-when-to-release-100

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing/InstrumentationWithActivitySource.cs
Comment (pakrym) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing/InstrumentationWithActivitySource.cs:15]: You shouldn't need this in a sample project

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing.csproj
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing.csproj]: Fixed

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs
Comment (pakrym) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs:13]: Is the connection string optional?

Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs:13]: No. We extract IKey from connection string. I defined all the properties, need to add logic for all the properties in next PRs.

**Comment by pakrym** [comment]
Can you please follow the https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/117/Setting-up-a-new-Unified-Pipeline to add a pipeline.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs:15]: Not sure why we need these options?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs:13]: can be internal. Every other exporter is internal in Otel main repo. Users get it through extension methods on Builder.

**Comment by reyang** [comment]
Adding @kamilon as FYI.

**Comment by reyang** [comment]
Adding @timr-dev as FYI.

**Comment by reyang** [comment]
Add @ThomsonTan as FYI.

**Comment by reyang** [comment]
Adding @CanFikret as FYI.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:17]: Why not take `this Activity activity` as parameter, to do full justice to the class name.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:98]: RequestTelemetry.statuscode should be coming from the tag "http.status_code".
RequestTelemetry.success should be calculated based on the tag "http.status_code".
Don't use StatusCode in short :) 
This can be addressed in separate PRs.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:17]: good point, will get his fixed.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:135]: minor: the method name should be something like GetHttpUrl

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:23]: It looks like the static use to be the ActivitySpanId object?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:25]: Would String.Empty be more logical?

**Review by cijothomas — APPROVED** [review]

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (TimothyMothra) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:159]: should this be `null` here? At line 144, you're only returning URL if it's != null.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:41]: switch/case seems like a better fit, especially if extended for other types (for readability). You also have the benefits of a jump table from switch.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (TimothyMothra) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:149]: minor: you could move two of the three `TryGetValue` calls into the if statement below. No need to do the extra gets if the Host is null. :)

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:159]: Yes. This will return `null`, this part needs to be revisited to align with OT Spec.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:76]: Would it have been logical to pre-calc the string instead of ToString, you have everything you need to do so without incurring a runtime hit/alloc.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:36]: Generally same comment as above, from an API design perspective unless String.Empty has a different meaning than null, you probably just want to default to string.empty, this way someone who's doing something like GetTraceId().ToLowerInvariant() won't get an unhappy surprise.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorExporterOptions.cs:13]: public setters gives you the problem where construction of the option doesn't enforce the restrictions/validation of the fields. So you end up with fun things like missing connection strings when they shouldn't be gone or someone changing it unexpectdely.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:41]: There is a TODO work, pending in this method. Will change this to switch in next PR.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs
Comment (ThomsonTan) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:58]: >StatusCanonicalCode.Unknown => 0, [](start = 16, length = 33)

Is this still needed as 0 is the default?

Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ActivityExtensions.cs:58]: we can completely ignore status for now - as OpenTelemetry is removing Status from spec.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs
Comment (timr-dev) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs:43]: This is a bit weird... do you plan on implementing it? Is cancellationToken.Cancel() insufficent?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs
Comment (ThomsonTan) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:6]: Mov these 3 to follow all `using System.*`?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporter.cs:43]: @reyang has simplified this part in OT. I have a nuget reference to beta version which is little old. This part of the code will change.

**Comment by rajkumar-rangaraj** [comment]
> Can you please follow the https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/117/Setting-up-a-new-Unified-Pipeline to add a CI pipeline.

@pakrym, Followed instructions and created ci.yml, could you please check.

**Review by pakrym — APPROVED** [review]

**Comment by chidozieononiwu** [comment]
This package was added without a README.md which now [brakes CI](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=503901&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=9069f0a2-bd31-51b1-8b5c-08355cf80229) . Could you consider adding a README.md following [this example](https://github.com/Azure/azure-sdk/blob/master/docs/policies/README-EXAMPLE.md). Or you can add your package name to the README `known_presence_issues` to skip the check for the README on CI. This is also a good time to start a CHANGELOG.mg file in the same directory using [this guidance](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#changelog-guidance)

**Pull request merged**
